routers/stops.py

```diff
@@ -1,66 +1,86 @@
+            if current_user.entity != UserEntity.admin:
                 raise HTTPException(
                     status_code=status.HTTP_403_FORBIDDEN,
-                    detail="Passengers can only request one stop at a time",
+                    detail="Only admins can create static stops",
                 )
-        # Dummy driver selection: first driver in the table
-        # Query the user table to get all drivers
-        response = supabase.table("users").select("id").eq("entity", UserEntity.driver.value).execute()
-        # Check if response has data
-        if response.data:
-            driver_id = response.data[0]["id"]
+            if stop.bus_id is None:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Static stop must specify a bus",
+                )
+            # TODO: (optional) if the stop already exists, only insert new bus mapping
         else:
-            # Raise an exception if the stop is not possible
-            # TODO: change to another status code, e.g., 2xx
-            raise HTTPException(
-                status_code=status.HTTP_400_BAD_REQUEST,
-                detail="Creating a stop is not possible",
-            )
+            # Reset the bus line if specified and determine it dynamically
+            stop.bus_id = None
+            nearest_stops = get_stops_sorted(lat=stop.lat, long=stop.long)["stops"]
+            for nearest_stop in nearest_stops:
+                nearest_stop_distance = nearest_stop["dist_meters"]
+                if nearest_stop_distance > 1000:
+                    break
+                nearest_stop_id = nearest_stop["id"]
+                response = supabase.table("bus_stop_mappings").select("bus_id").eq("stop_id", nearest_stop_id).execute()
+                # Check if response has data
+                if response.data:
+                    # Take the bus line that comes first in the table
+                    # TODO: (optional) add more complex logic here
+                    nearest_bus_id = response.data[0]["bus_id"]
+                else:
+                    # No buses: try next closest stop
+                    continue
+                # TODO: increase afterwards
+                if nearest_stop_distance < 10:
+                    # Return existing stop info
+                    return {"bus_id" : nearest_bus_id, **nearest_stop}
+                else:
+                    stop.bus_id = nearest_bus_id
+                    break
+            if stop.bus_id is None:
+                # Raise an exception if the stop is not possible
+                # TODO: (optional) change to another status code, e.g., 2xx
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Stop creation is not possible: no bus lines nearby",
+                )
         # Insert the stop data into the stop table
```

Comment:
Looks rather complicated, we should ideally derive this functionality into functions